tinygrad/shape/symbolic.py

```diff
@@ -137,48 +142,49 @@ class NumNode(Node):
   def __init__(self, num:int):
     self.b, self.min, self.max = num, num, num
   def __int__(self): return self.b
+  def __eq__(self, other): return self.b == other
+  def __hash__(self): return self.hash  # needed with __eq__ override
 
 def create_node(ret:Node):
   assert ret.min <= ret.max, f"min greater than max! {ret.min} {ret.max} when creating {type(ret)} {ret}"
-  if ret.min == ret.max: return NumNode(ret.min)
+  if ret.min == ret.max: return NumNode(ret.min) if isinstance(ret.min, int) else ret.min
```

Comment:
Huh? How is ret.min not an int?

Reply:
with symbolic stride, idx can have symbolic max, and a node derived from that can have symbolic min, max

Reply:
oh...this is never how max was intended to work. is this a requirement?

Reply:
I misunderstood the root cause. The symbolic max for an idx is still needed, and we will later update renderer to render the max in for loop too.

The root cause was that `NumNode * Node` turned into `NumNode(Node)`, and the correct fix is to unwrap in `__mul__`. I have updated the PR and the tests accordingly.